src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -0,0 +1,16 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Pose3d;
+
+public interface VisionSubsystem {
```

Comment:
I think that you will always want a `VisionSubsystem` to extend `SubsystemBase`. If so, you might consider making this an abstract class instead of an interface, and extend `SubsystemBase` here. E.g.:

```suggestion
public abstract class VisionSubsystem extends SubsystemBase {
```

You would then need to make the method definitions `abstract`, e.g. `public abstract double getTurretYaw(int id);`.

Not a huge change and not required, but I think it simplifies things because you can use `VisionSubsystem` as a variable type, instead of having to switch things out depending on which one has been instantiated.

Reply:
I made a card in the backlog for it, and we can discuss it more on Saturday because I'm not entirely sure how abstract classes work

Reply:
It's pretty easy; I think if we sat down together we could convert it in about 10 minutes.